### ISSUE TITLE
Unsuitable function names for loading configs

### ISSUE THREAD

**Issue by clessig** [ecmwf/WeatherGenerator#218]
config: `load_model_config()` and `load_config()` are misleading names. 

`load_model_config` : contains model *and* run parameters, e.g. related to learning rate, #epochs etc. So an appropriate name would be load_run_config. It's a good idea to separate the config into a model config and training config but it's not there yet.

`load_config` : this loads and merges the various configs; maybe a good name would be `load_merge_configs`. The function is also missing documentation. In `load_config`, epoch and overwrite_config should also have default arguments `None`.

CC: @grassesi , @tjhunter
